### ISSUE TITLE
Align docstring

### ISSUE THREAD

**Issue by SevereOverfl0w** [kkinnear/zprint#28]
Using the docstring detection, would it be possible to perform formatting to match https://github.com/bbatsov/clojure-style-guide#align-docstring-lines ?

**Comment by kkinnear** [comment]
Thanks for the suggestion!  It is always interesting to know what people are looking for.

It would help me to know in more detail what you are asking so that I know whether or not what I can do would in fact be useful to you.

I could certainly ensure that there is a minimum indent, and turn this:
```clojure
(defn foo
  "Foo does this and that, and sometimes
 with the right inputs, does other things too."
  [a b c]
  ...)
```
into this:
```clojure
(defn foo
  "Foo does this and that, and sometimes
  with the right inputs, does other things too."
  [a b c]
  ...)
```
However, going the other direction, there is a problem distinguishing this:
```clojure
(defn foo
  "Foo does this and that, and sometimes
            with the right inputs, does other things too."
  [a b c]
  ...)
```
from this
```clojure
(defn foo
  "Foo does this and that, and sometimes
  with the right inputs, does other things too.
    Inputs:
      a     good stuff
      b     bad stuff
      c     stuff
    Returns:
      Some things that you wouldn't believe"
  [a b c]
  ...)
```
so if I try to move things from the right over to the left, I can end up with this:
```clojure
(defn foo
  "Foo does this and that, and sometimes
  with the right inputs, does other things too.
  Inputs:
  a     good stuff
  b     bad stuff
  c     stuff
  Outputs:
  Some things that you wouldn't believe"
  [a b c]
  ...)
```
which in my view messes up the original authors good intentions in order to make a change that probably is rarely necessary.

So, while it is a certain amount of work (some of which i've already done to put specs into the docstrings), if you would be satisfied with the original "ensure that there is a minimal ident" I would be happy to put that on the list of things to implement.  If you wanted more radical transformations, then I'd like to know specifically what you have in mind (and how you imagine that I might distinguish when the more radical transformation were justified).  Some examples would be helpful to illustrate what you have in mind.

Thanks again for the suggestion!

**Comment by SevereOverfl0w** [comment]
You're right, this is difficult to do, particularly as there's so few standards around this. I think a minimum indent would be helpful, some people simply forget. I do sometimes see alignment like this too:

```
(defn x
  "Foo
   Bar"
  [baz] …)
```

I'm wondering if detection for the case where all the lines have an additional indentation of +1 would be valuable also.

**Comment by kkinnear** [comment]
Sounds reasonable, I think I could figure out how to remove the +1 indentation (and let +2 and greater stay).  It may be a while before I can get to this, but I'll put it on the to-do list.  Thanks for the idea!

**Comment by SevereOverfl0w** [comment]
Thanks for taking the time to consider it


Dominic

On 13 Jun 2017 3:25 am, Kim Kinnear <notifications@github.com> wrote:

Sounds reasonable, I think I could figure out how to remove the +1 indentation (and let +2 and greater stay). It may be a while before I can get to this, but I'll put it on the to-do list. Thanks for the idea!
